framework/src/executioners/Executioner.C

```diff
@@ -103,3 +110,69 @@ Executioner::addAttributeReporter(const std::string & name, Real initial_value)
 
   return value;
 }
+
+void
+Executioner::addSolveObject(const std::string & type, const std::string & name,
+                            InputParameters & parameters)
+{
+  if (_solve_object_names.count(type))
+    mooseError("Solve object in type ", type, " has been constructed");
+
+  parameters.set<SubProblem *>("_subproblem") = &feProblem();
+  parameters.set<SystemBase *>("_sys") = &feProblem().getNonlinearSystemBase();
+  parameters.set<bool>("verbose") = getParam<bool>("verbose");
+  _solve_objects[name] = _app.getFactory().create<SolveObject>(type, name, parameters, 0);
+  _solve_object_names[type] = name;
+}
+
+void
+Executioner::init()
+{
+  checkIntegrity();
+  _fe_problem.execute(EXEC_PRE_MULTIAPP_SETUP);
+  _fe_problem.initialSetup();
+}
+
+void
+Executioner::execute()
+{
+  if (!_head_solve_object)
+    mooseError("Executioner has not been set up for executioner with 'solver_object_ordering'");
```

Comment:
I would want the other executioners to use this too.
All below is duplication basically

Reply:
I have not thought over this very carefully. This is mainly related with outputting and more execute_on flags. I do not think this code is final. Ideally we just need to call `_head_solve_object->solve()` here. But without improving outputting and execute_on flag, which possibly should be part of solve objects, we cannot go further here imho.

Reply:
I possible should handle `_time_step` more carefully and move the output on initial into `preExecute` and output on final into `postExecute`. Note that we do not have time stepping solve object currently, so we can be lazy on setting `_time_step`.

Reply:
I changed the code back and forth here and think it is still a long way to make `Executioner::execute` great mainly because of the outputting and execute_on and lack of some major solve objects. I can remove `Executioner::execute` and the tests if you want. I do not need this in Griffin but just the machinery of handling solve objects. What do you think?

Reply:
I want steady, transient and eigenvalue to be able to use the solve object nesting, and I dont want an alternative general execute unless we can prove that we need this.